### PR TITLE
chore: make 0.3.0 the latest bundle version

### PR DESCRIPTION
**Pull request by jopit**
Signed-off-by: John Pitman <jpitman@redhat.com>

**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

> /kind chore

**What does this PR do / why we need it**:

PR https://github.com/argoproj-labs/argocd-operator/pull/563 updated the latest bundle version to be 0.3.0.  After discussion with Abhishek and William, this is the wrong thing to do, the latest bundle version should be 0.3.0 as 0.3 has not yet been released.  This PR fixes that.

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

### CODE REVIEW

File: bundle/manifests/argocd-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -1048,4 +1048,4 @@ spec:
   provider:
     name: Argo CD Community
   replaces: argocd-operator.v0.1.0
```
Comment (iam-veeramalla) [inline bundle/manifests/argocd-operator.clusterserviceversion.yaml:1050]: it should be replacing `argocd-operator-v0.2.0` ?

Comment (jopit) [reply bundle/manifests/argocd-operator.clusterserviceversion.yaml:1050]: Thanks, I missed that.

File: config/manifests/bases/argocd-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -660,5 +660,5 @@ spec:
   maturity: alpha
   provider:
     name: Argo CD Community
-  replaces: argocd-operator.v0.1.0
-  version: 0.2.0
+  replaces: argocd-operator.v0.2.0
+  version: 0.0.0
```
Comment (iam-veeramalla) [inline config/manifests/bases/argocd-operator.clusterserviceversion.yaml:664]: should be 0.3.0 ?

Comment (jopit) [reply config/manifests/bases/argocd-operator.clusterserviceversion.yaml:664]: It doesn't matter, it gets replaced in the generated CSV by the correct version.

**Review by iam-veeramalla — APPROVED** [review]
LGTM, Thanks @jopit

**Pull request merged**